File: coredata/ObjC/_NotSameWrapper.m
#import "_NotSameWrapper.h"
#import "NotSame.h"

@interface _NotSameWrapper ()
@end

@implementation _NotSameWrapper {
}
- (instancetype)initWithEntity:(NotSame *)entity {
    self = [super init];
    if (self) {
        _entity = entity;
        self.boolean = entity.boolean;
        self.data = entity.data;
        self.date = entity.date;
        self.decimal = entity.decimal;
        self.dou = entity.dou;
        self.flo = entity.flo;
        self.int16 = entity.int16;
        self.int32 = entity.int32;
        self.int64 = entity.int64;
        self.string = entity.string;
        self.transformable = entity.transformable;
    }
    return self;
}

- (void)updateEntity:(NotSame *)entity {
    entity.boolean = self.boolean;
    entity.data = self.data;
    entity.date = self.date;
    entity.decimal = self.decimal;
    entity.dou = self.dou;
    entity.flo = self.flo;
    entity.int16 = self.int16;
    entity.int32 = self.int32;
    entity.int64 = self.int64;
    entity.string = self.string;
    entity.transformable = self.transformable;
}
@end

File: coredata/Swift/_NotSameWrapper.swift
import Foundation

class _NotSameWrapper: NSObject {
    let _entity: NotSame?
    func entity() -> NotSame? {
        return _entity
    }
    var boolean: NSNumber?
    var data: NSData?
    var date: NSDate?
    var decimal: NSDecimalNumber?
    var dou: NSNumber?
    var flo: NSNumber?
    var int16: NSNumber?
    var int32: NSNumber?
    var int64: NSNumber?
    var string: String?
    var transformable: AnyObject?

    init(notSame: NotSame?) {
        _entity = notSame
        super.init()
        if let e = notSame {
            self.boolean = e.boolean
            self.data = e.data
            self.date = e.date
            self.decimal = e.decimal
            self.dou = e.dou
            self.flo = e.flo
            self.int16 = e.int16
            self.int32 = e.int32
            self.int64 = e.int64
            self.string = e.string
            self.transformable = e.transformable
        }
    }
    func updateNotSame(notSame: NotSame) {
        notSame.boolean = self.boolean
        notSame.data = self.data
        notSame.date = self.date
        notSame.decimal = self.decimal
        notSame.dou = self.dou
        notSame.flo = self.flo
        notSame.int16 = self.int16
        notSame.int32 = self.int32
        notSame.int64 = self.int64
        notSame.string = self.string
        notSame.transformable = self.transformable
    }
}
